### DISCUSSION TITLE
Bring back `data-test` attributes?

### DISCUSSION THREAD

**Discussion by mrnerdhair** [shapeshift/web#1477] [General]
I think we should revert https://github.com/shapeshift/web/pull/849/commits/294045872c5c100e3e1f13277b7c0125753383e0.
- I don't think the bandwidth savings aren't very significant compared to adding a dependency on `babel-plugin-react-remove-properties` and more logic to `react-app-rewired.config.js`.
- We don't care if bots automate the site -- they'd just hit our APIs if they felt malicious
- I think the `data-test` tags would be useful in alpha for attaching Pendo guides to UI elements, as well as increasing the utility of its auto-generated analytics events.

@rbeer @0xApotheosis thoughts?

**Comment by 0xApotheosis** [comment]
I'm totally ok with doing this. 

The attribute removal was only done as a nicety, and if there are additional benefits for Pendo by having them in the build that's a win.

I'll approve a PR that reverts https://github.com/shapeshift/web/commit/294045872c5c100e3e1f13277b7c0125753383e0.

**Comment by 0xdef1cafe** [comment]
i'm on board with this, they'll help with Pendo and cypress tests

**Comment by GMSteuart** [comment]
Alternatively, another solution would be to utilize the default [webpack code splitting based off mode](https://webpack.js.org/guides/production/#specify-the-mode) and add or remove props based off `process.env.NODE_ENV`

A simplified example:

```ts
const props = { src: 'https://github.com'}
if (process.env.NODE_ENV !== 'production') {
   Object.assign({}, props, { 'data-test': 'my-component' }
 } 
// ...
```
The code block for the `if` statement will not be in the production file output and so the attribute will only exist in non-production builds.

**Comment by 0xApotheosis** [comment]
For posterity, we brought back `data-test` attributes in https://github.com/shapeshift/web/pull/1660.

I'll go ahead and lock this conversation.
